azure/defaults.go

```diff
@@ -84,6 +84,14 @@ const (
 	ProviderIDPrefix = "azure://"
 )
 
+const (
+	// CustomHeaderPrefix is the prefix of annotations that enable additional cluster / node pool features.
+	// Whatever follows the prefix will be passed as a header to cluster/node pool creation/update requests.
+	// E.g. add `"custom-headers.azure.dev/UseGPUDedicatedVHD": "true"` annotation to AzureManagedMachinePool CR
+	// to enable creating GPU nodes by the node pool.
+	CustomHeaderPrefix = "custom-headers.azure.dev/"
```

Comment:
nit: 
Do annotation/labels follow a specific pattern?  In case of annotations I have seen `cluster.x-k8s.io` in CAPI. May be worth discussing as  his will help us being consistent while we may introduce more annotations in near future. WDYT?

Reply:
Definitely. Do you suggest opening a thread in `Disussions` or should we do it here? Do you have any suggestions? Maybe we can use `cluster-azure.x-k8s.io/` for all project annotations and `cluster-azure.x-k8s.io/custom-header-{HEADER_NAME}` specifically for this feature?

Reply:
`cluster-azure.x-k8s.io` -- This one sounds good to me. Copying @CecileRobertMichon @shysank to provide more input.

Reply:
how about `infrastructure.cluster.x-k8s.io`? `cluster-azure.x-k8s.io` is the API Group for Cluster API. CAPZ resources use `infrastructure.cluster.x-k8s.io` as its Group.


https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2349883d1effe6344690459b40e5dc26138010df/api/v1beta1/groupversion_info.go#L29

Reply:
@CecileRobertMichon makes sense, thanks. I've changed it.